libff/algebra/curves/tests/test_groups.cpp

```diff
@@ -26,6 +26,19 @@
 
 using namespace libff;
 
+// Utility function to compute a point on the curve E(Fq) with the given x
+// coordinate.
+template<typename GroupT>
+GroupT curve_point_at_x(const typename GroupT::base_field &x)
+{
+    const typename GroupT::base_field x_squared = x * x;
+    const typename GroupT::base_field x_cubed = x_squared * x;
+    const typename GroupT::base_field y_squared =
+        x_cubed + (GroupT::coeff_a * x_squared) + GroupT::coeff_b;
+    const typename GroupT::base_field y = y_squared.sqrt();
```

Comment:
As far as I remember, the `sqrt` function does not terminate if the squared element is not a quadratic residue in the field. Hence, this will cause troubles down the line. Let's check the Legendre symbol/ Euler criterion of the `y_squared` element before calling `sqrt`.

Reply:
Added the check.

Reply:
Thanks